src/components/Calculator.jsx

```diff
@@ -1,77 +1,60 @@
-import React from 'react';
+import React, { useState } from 'react';
 import Key from './InputCalc';
 import calculate from './logic/calculate';
 
-class Calculator extends React.PureComponent {
-  constructor(props) {
-    super(props);
-    this.retrieveSymbol = this.retrieveSymbol.bind(this);
-    this.state = {
-      total: null,
-      next: null,
-      operation: null,
-    };
-  }
+const Calculator = () => {
+  const [total, setTotal] = useState(null);
+  const [next, setNext] = useState(null);
+  const [operation, setOperation] = useState(null);
+  const [display, setDisplay] = useState('0');
 
-  retrieveSymbol(symbol) {
-    const {
-      total, next, operation,
-    } = this.state;
-    console.log(symbol);
+  const retrieveSymbol = (symbol) => {
+    const calcObj = calculate({ total, next, operation }, symbol);
+    setTotal(calcObj.total);
+    setNext(calcObj.next);
+    setOperation(calcObj.operation);
 
-    this.setState(calculate({ total, next, operation }, symbol));
-    this.setState((prevState) => ({
-      total: prevState.total,
-      next: prevState.next,
-      operation: prevState.operation,
-    }));
+    if (!calcObj.next && !calcObj.total && !calcObj.operation) { setDisplay('0'); }
+    if (calcObj.next && !calcObj.total && !calcObj.operation) { setDisplay(calcObj.next); }
+    if (!calcObj.next && calcObj.total && calcObj.operation) { setDisplay('0'); }
+    if (calcObj.next && calcObj.total && calcObj.operation) { setDisplay(calcObj.next); }
+    if (!calcObj.next && calcObj.total && !calcObj.operation) { setDisplay(calcObj.total); }
+  };
```

Comment:
- [ ] OPTIONAL: This could have been a separate component (`Buttons`) in a separate file and rendered here. Smaller components are easy to fix, reusable, and make the app more scalable. :pray: